### PR TITLE
Chain Selection: Follow-ups

### LINKED ISSUE #3293: Implement remainder of Chain Selection Subsystem
#3277 got quite large, so I decided to break out some smaller items into a follow-up pull request.

 - Plug into overseer
 - Implement DB
 - Send Approve messages from Approval Voting (even when insta-approving)
 - Stagnation checking loop

### PR DESCRIPTION
**Pull request by rphmeier**
This implements the follow-ups to #3277 and closes #3293 .

I don't yet plug it into overseer as that'll take a DB migration, for which we should do the migration of dispute coordinator at the same time. I'd like to wait until #2581 is implemented before that, just in case there are further changes we want to make to the coordinator.

So this PR:
  - [x] Implements the `DbBackend` fulfilling the `Backend` trait.
  - [x] Implements the stagnation checking loop
  - [x] Configures Approval Voting to send `ChainSelectionMessage::Approved`
  - [x] Tests the database backend on an in-memory KVDB.
  - [x] Tests stagnation using a mockable clock

and is ready for review when these checkboxes are all done.

### CODE REVIEW

File: node/core/chain-selection/src/lib.rs [hunk 1]
```diff
@@ -202,96 +209,143 @@ impl Error {
...
+		let mut delay = futures_timer::Delay::new(interval);
+
+		futures::stream::poll_fn(move |cx| {
+			let poll = delay.poll_unpin(cx);
+			if poll.is_ready() {
+				delay.reset(interval)
+			}
+
+			poll.map(Some)
+		})
```
Comment (ordian) [inline node/core/chain-selection/src/lib.rs:279]: what about reusing https://github.com/paritytech/polkadot/blob/f560edf6b40e18722d16512cef0e1e0a78eae08c/node/core/pvf/src/host.rs#L601-L609

which I think does the same, but is more readable

**Review by ordian — APPROVED** [review]

File: node/core/chain-selection/src/lib.rs (hunk 1 above)
Comment (rphmeier) [reply node/core/chain-selection/src/lib.rs:279]: What we're really missing here is a `futures::stream::poll_fn_async`. I don't think that DRY is accurate in this case; a 'pulse_every' floating in a random util crate is worse than a small amount of code reuse.

File: node/core/chain-selection/src/db_backend/v1.rs [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+			&*self.inner,
+			self.config.col_data,
+			&stagnant_at_key(timestamp.into()),
+		).map(|o| o.unwrap_or_default())
+	}
+
+	fn load_stagnant_at_up_to(&self, up_to: crate::Timestamp)
+		-> Result<Vec<(crate::Timestamp, Vec<Hash>)>, Error>
+	{
+		let stagnant_at_iter = self.inner.iter_with_prefix(
```
Comment (ordian) [inline node/core/chain-selection/src/db_backend/v1.rs:246]: just a note, that this adds another reliance on `iter_with_prefix` which needs to be removed if we're to migrate off `kvdb` interface

Comment (rphmeier) [reply node/core/chain-selection/src/db_backend/v1.rs:246]: Right. I think that's fine for now based on discussions in Element

**Pull request merged**